experimental/InvariantTheory/AlgebraicGroups.jl

```diff
@@ -0,0 +1,218 @@
+
+Constructs the special linear group of rank `n` over the field `k`.
+"""
+function AlgGroup_SL(n::Int, k::FieldType) where{FieldType<:Field}
+  R, z = PolynomialRing(k, "z" => (1:n, 1:n))
+  Z = matrix(z)
+  I = ideal(R, [1-det(Z)] )
+  return AlgGroup( k, R, I, Z )
+end
+  
+
+
+function subgroup( G::AlgGroup, I::MPolyIdeal )
+  @assert parent(I) == parent(G.R) "given ideal does not live in the same ring as the defining ideal of the algebraic group"
+  return AlgGroup( G.R, G.I + I, G.Z )
+end
+
+@Markdown.doc """
+    LinearRepresentation{FieldType, RingType, RingElemType, MatrixTypeG, MatrixTypeA}
+
+A linear representation of an algebraic group G ⊂ GL(t,𝕜) over a field 𝕜, 
+defined on a vector space V ≅ 𝕜ʳ. 
+The information stored consists of 
+  * an instance `G` of `AlgGroup{FieldType, RingType, RingElemType, MatrixTypeG}` 
+    which describes the group as a subscheme of GL(t,𝕜);
+  * an r×r-matrix A of type `MatrixTypeA` with entries in the polynomial ring 𝕜[zᵢⱼ| 1 ≤ i,j ≤ t],
+    whose variables are the standard coordinates on GL(t,𝕜), inducing the 
+    map G →  GL(r,𝕜) ≅ End(V) for the representation.
+"""
+mutable struct LinearRepresentation{FieldType, RingType, RingElemType, MatrixTypeG, MatrixTypeA}
+  G::AlgGroup{FieldType, RingType, RingElemType, MatrixTypeG}
+  A::MatrixTypeA
+  pullback_morphism::Oscar.AlgHom
+  #pullback_morphism::AlgebraHomomorphism
+
+  function LinearRepresentation( G::AlgGroup, A::MatSpaceElem )
+    @assert G.k === base_ring(base_ring(A))
+    @assert ncols(G.Z)*nrows(G.Z) == length(gens(base_ring(A)))
+    images = Vector{elem_type(ambient_ring(G))}()
+    Z = presentation_matrix(G)
+    n = ncols(Z)
+    # calling gens(base_ring(A)) reveals that the order of the 
+    # coordinates is reversed...
+    for j in (1:n)
+      for i in (1:n)
+	push!(images, Z[i,j])
+      end
+    end
+    phi = AlgebraHomomorphism(base_ring(A), ambient_ring(G), images)
+    return new{typeof(G.k), typeof(G.R), elem_type(G.R), typeof(G.Z), typeof(A)}(G, A, phi)
+  end
+
+end
+
+# the identity representation
+function LinearRepresentation( G::AlgGroup )
+  n = ncols(G.Z)
+  S, a = PolynomialRing(base_field(G), "a" => (1:n, 1:n))
+  A = matrix(a)
+  return LinearRepresentation( G, A )
+end
+
+@Markdown.doc """
+    mapping_matrix( rep::LinearRepresentation )
+
+Returns the r×r-matrix A with polynomial entries in the 
+variables zᵢⱼ, 1≤i,j≤n of the group G.
+"""
+function mapping_matrix( rep::LinearRepresentation )
+  return rep.A
+end
+
+@Markdown.doc """
+    group( rep::LinearRepresentation )
+
+Returns the group G which is represented by `rep`.
+"""
+function group( rep::LinearRepresentation )
+  return rep.G
+end
+
+@Markdown.doc """
+    representation_on_sym_power( rep::LinearRepresentation, d::Int )
+
+Constructs the induced representation on the d-th symmetric power 
+Symᵈ V* of the vector space V on which `rep` is defined.
+"""
+function representation_on_sym_power( rep::LinearRepresentation, d::Int )
```

Comment:
How about `symmetric` instead of `sym`? Saving 6 characters but introducing seemingly random abbreviations. With tab-completion the length does not matter at all.